addons/account/models/account_move.py

```diff
@@ -3381,9 +3381,13 @@ def default_get(self, default_fields):
 
             # Suggest default value for 'partner_id'.
             if 'partner_id' in default_fields and not values.get('partner_id'):
-                partners = move.line_ids[-2:].mapped('partner_id')
-                if len(partners) == 1:
-                    values['partner_id'] = partners.id
+                if len(move.line_ids[-2:]) == 2 and  move.line_ids[-1].partner_id == move.line_ids[-2].partner_id:
+                    values['partner_id'] = move.line_ids[-2:].mapped('partner_id').id
```

Comment:
what happens if both lines don't have a partner?

Reply:
@MiquelRForgeFlow  No partner => No suggestion because move.line_ids[-2:].mapped('partner_id').id will return False

Reply:
Ok. Well, my point was trying to ask the convenience or not about passing `values['partner_id'] = False` in this use case (because before this patch the `values['partner_id'] = False` was never passed). I think that passing it or not passing it doesn't produce any difference, but I was asking just to confirm it.